### PR TITLE
move apimachinery/announced to apimachinery

### PR DESCRIPTION
**Pull request by deads2k**
This also rewires the announce package to allow installation into multiple groups via multiple calls to `install.Install`.  At some point, we'll want to unwire the hardcoded, "install to this scheme" call.

@lavalamp something we've wanted for a while and need for genericapiserver
@seh you've asked related questions
@sttts ptal.  First commit is interesting, second commit is a straight move.

### CODE REVIEW

**Comment by deads2k** [comment]
> @deads2k PR needs rebase

wow, that might be a record.

File: pkg/genericapiserver/server/genericapiserver_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func TestInstallAPIGroups(t *testing.T) {
 			}, nil
 		}
 
-		mapper := api.NewDefaultRESTMapperFromScheme([]schema.GroupVersion{gv}, interfacesFor, "", sets.NewString(), sets.NewString(), scheme)
```
Comment (sttts) [inline pkg/genericapiserver/server/genericapiserver_test.go]: Why do we still have NewDefaultRESTMapperFromScheme in pkg/api in the main repo?

File: staging/src/k8s.io/apimachinery/pkg/api/meta/default.go [hunk 2]
```diff
@@ -14,28 +14,21 @@ See the License for the specific language governing permissions and
...
 )
 
-// Instantiates a DefaultRESTMapper based on types registered in api.Scheme
-func NewDefaultRESTMapper(defaultGroupVersions []schema.GroupVersion, interfacesFunc meta.VersionInterfacesFunc,
-	importPathPrefix string, ignoredKinds, rootScoped sets.String) *meta.DefaultRESTMapper {
-	return NewDefaultRESTMapperFromScheme(defaultGroupVersions, interfacesFunc, importPathPrefix, ignoredKinds, rootScoped, Scheme)
-}
-
-// Instantiates a DefaultRESTMapper based on types registered in the given scheme.
-func NewDefaultRESTMapperFromScheme(defaultGroupVersions []schema.GroupVersion, interfacesFunc meta.VersionInterfacesFunc,
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/default.go]: ... answering my previous question. Here it goes.

**Comment by sttts** [comment]
Don't see anything that speaks against making registry and scheme explicit in every group. Looks good.

**Pull request merged**